utils/gitlab_api.py

```diff
@@ -0,0 +1,97 @@
+import gitlab
+import urllib3
+import uuid
+from reconcile.config import get_config
+
+
+class GitLabApi(object):
+    def __init__(self, project_name, ssl_verify=False):
+        # The following line will supress
+        # `InsecureRequestWarning: Unverified HTTPS request is being made`
+        urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+        self.project = init_project_from_config(project_name, ssl_verify)
+
+    def create_branch(self, new_branch, source_branch):
+        data = {
+            'branch': new_branch,
+            'ref': source_branch
+        }
+        self.project.branches.create(data)
+
+    def delete_file(self, branch_name, file_path, commit_message):
+        data = {
+            'branch': branch_name,
+            'commit_message': commit_message,
+            'actions': [
+                {
+                    'action': 'delete',
+                    'file_path': file_path
+                    }
+            ]
+        }
+        self.project.commits.create(data)
+
+    def create_mr(self, source_branch, target_branch, title,
+                  remove_source_branch=True):
+        data = {
+            'source_branch': source_branch,
+            'target_branch': target_branch,
+            'title': title,
+            'remove_source_branch': str(remove_source_branch)
+        }
+        self.project.mergerequests.create(data)
+
+    def delete_user_mr_exists(self, title):
+        mrs = self.project.mergerequests.list(state='opened')
+        for mr in mrs:
+            # since we are using a naming convention for these MRs
+            # we can determine if a pending MR exists based on the title
+            if (mr.attributes['title'] != title):
```

Comment:
unnecessary outer paens

Reply:
also, perhaps you should do mr.attributes.get('title') instead, who knows if GLApi always returns that...